[receiver/mysql] Pass pointer to ScrapeErrors instead of by value

BinaryFissionGames opened the pull request:
**Description:** <Describe what has changed.>
* fixes issue where partial errors are not recorded due to passing the ScrapeErrors struct by value
* rename errors -> errs to avoid possible shadowing of the errors package

**Link to tracking Issue:** #13009

**Testing:**
* Added a test with a partial scrape error to test that partial errors are being recorded

**Documentation:** N/A

BinaryFissionGames committed:
pass scrapererrs as pointer

BinaryFissionGames committed:
small test name consistency thing

BinaryFissionGames committed:
add changelog

schmikei commented:
Just at a glance but is the `receiver/mysqlreceiver/testdata/scraper/empty.txt` used for anything?

BinaryFissionGames committed:
make empty.txt purpose more clear

BinaryFissionGames commented:
> Just at a glance but is the `receiver/mysqlreceiver/testdata/scraper/empty.txt` used for anything?
It's used as "partial data" for testing, I renamed it and added a header to make it more clear what it's meant to be used for.
